### PR TITLE
Add Ledger Nano S support

### PR DESCRIPTION
**Pull request by lenondupe**
I've developed an app for the Ledger Nano S hardware wallet that will be released in the Ledger store this week. This patch adds the ability to use Stellarterm in conjunction with that app. On the account page you can sign in using the Ledger. The sign in button gets enabled when the user connects the Ledger and opens the Stellar app. Transaction signing then happens on the Ledger device. Whenever an action is done that requires a transaction to be signed the user is prompted with information about the transaction on the device and is requested to approve the transaction.

Note: to test this the application needs to be served over https as required by u2f standard.

### CODE REVIEW

**Comment by lenondupe** [comment]
I see travis build is failing. I found this: https://github.com/node-hid/node-hid/issues/31 
It suggests `libusb-dev` might not be installed on the build server.

**Comment by irisli** [comment]
Cool! I'll look at this soon

**Comment by irisli** [comment]
I think before I want to pull these changes in, I want to add a read-only mode to StellarTerm.

**Comment by lenondupe** [comment]
Not sure I follow. When would you sign something if not absolutely necessary? Or are you thinking of batching multiple operations in a single transaction?

**Comment by irisli** [comment]
Oh what I mean is:

Right now, the UX flow is that the user enters their passphrase or passphrase or whatever. Then, they have unlimited access. I'm thinking to add a read-only mode where the user first enters their public key. Then, they can make a transaction and will only need to insert secret key or use Ledger when finally necessary.

**Comment by lenondupe** [comment]
Right, that makes sense. That would make the user interaction more uniform.

**Comment by irisli** [comment]
I'll work on this once I have more time soon

**Comment by irisli** [comment]
Oh I just realized that there is still another problem! Someone might have multiple accounts on their Ledger nano. So the read-only option should be in addition to the logging in with Ledger Nano.

I'm thinking that in the login flow, users can pick to enter their account in any way:
- Read mode (Can authenticate with any other method later)
- Secret key
- Ledger
- other methods

**Comment by lenondupe** [comment]
I have been helping the developer of lupoex.com integrate the support as well and the way he does it he has three methods to sign in: 1. public key; 2. secret key; 3. ledger. When you sign in with ledger you can specify the account (bip32 path). Then when it is time to sign a transaction you are again shown 3 options: 1. manual; 2. secret key; 3. ledger. Manual signing means you are presented with the xdr and you can copy it and sign it in the way you want maybe on an air-gapped computer or something. When you choose secret key and you logged in with the secret key you have the option to use that one, but you get a field to specify a secret key either way, so you can enter a different one. When you choose to sign with ledger you can only sign with the account with which you logged in.

**Comment by irisli** [comment]
To get this merged in, I need to redo the transaction submission flow to be asynchronous.

I recently bought a Ledger Nano S and will work on this when I have time soon.

**Comment by irisli** [comment]
Closing this PR for the new one

File: src/components/Session/LoginForm.jsx [hunk 1]
```diff
@@ -41,6 +59,20 @@ export default class LoginForm extends React.Component {
...
       });
     }
+    this.proceedWithLedger = (event) => {
+      event.preventDefault();
+      if (!isValidBip32Path(this.state.bip32Path)) {
+        return this.setState({
+          invalidBip32Path: true
+        });
+      }
+      props.handler(null, true, this.state.bip32Path);
```
Comment (irisli) [inline src/components/Session/LoginForm.jsx]: props should never be mutated from inside a component

**Pull request closed without merging**